packages/ui/components/ui/tx/view/value.tsx

```diff
@@ -5,17 +5,71 @@ import {
   getDisplayDenomFromView,
 } from '@penumbra-zone/types';
 import { CopyToClipboard } from '../../copy-to-clipboard';
+import { AssetIcon } from './asset-icon';
 import { CopyIcon } from '@radix-ui/react-icons';
 import { Amount } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/num/v1/num_pb';
 
 interface ValueViewProps {
   view: ValueView | undefined;
   showDenom?: boolean;
+  showValue?: boolean;
+  showIcon?: boolean;
+  showEquivalent?: boolean;
 }
 
-export const ValueViewComponent = ({ view, showDenom = true }: ValueViewProps) => {
+export const ValueViewComponent = ({
+  view,
+  showDenom = true,
+  showValue = true,
+  showIcon = true,
+  showEquivalent = true,
+}: ValueViewProps) => {
   if (!view) return <></>;
 
+  if (view.valueView.case === 'knownAssetId' && view.valueView.value.metadata) {
+    const value = view.valueView.value;
+    const metadata = view.valueView.value.metadata;
+    const amount = value.amount ?? new Amount();
+    const exponent = getDisplayDenomExponent(metadata);
+    // The regex trims trailing zeros which toFormat adds in
+    const formattedAmount = fromBaseUnitAmount(amount, exponent)
+      .toFormat(6)
+      .replace(/(\.\d*?[1-9])0+$|\.0*$/, '$1');
+    const symbol = metadata.symbol || 'Unknown Asset';
+
+    return (
+      <div className='flex items-center text-base font-bold'>
+        {showIcon && <AssetIcon metadata={metadata} />}
+        {showValue && <span className='ml-1'>{formattedAmount}</span>}
+        {showDenom && <span className='ml-1'>{symbol}</span>}
+        {
+          // TODO: this will need refinement once we actually hand out
+          // equivalent values to the frontend. it would be nice to have
+          // another parameter that controls whether the valueview should
+          // fill width or not (with value to the left, equiv values to the right)
+        }
+        {showEquivalent &&
+          value.equivalentValues.map((equivalentValue, index) => {
+            const exponent = getDisplayDenomExponent(equivalentValue.numeraire);
+            const formattedEquivalent = fromBaseUnitAmount(
+              equivalentValue.equivalentAmount ?? new Amount(),
+              exponent,
+            )
+              .toFormat(6)
+              .replace(/(\.\d*?[1-9])0+$|\.0*$/, '$1');
+            const symbol = equivalentValue.numeraire?.symbol || 'Unknown Asset';
```

Comment:
the linter is complaining but i did this on purpose because it is the correct behavior here

Reply:
```suggestion
            // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
            const symbol = equivalentValue.numeraire?.symbol || 'Unknown Asset';
```

Reply:
tyvm!